cloud/misc/virt_net.py

```diff
@@ -0,0 +1,557 @@
+    uri:
+        required: false
+        default: "qemu:///system"
+        description:
+            - libvirt connection uri.
+    xml:
+        required: false
+        description:
+            - XML document used with the define command.
+requirements:
+    - "python >= 2.6"
+    - "python-libvirt"
+    - "python-lxml"
+'''
+
+EXAMPLES = '''
+# Define a new network
+- virt_net: command=define name=br_nat xml='{{ lookup("template", "network/bridge.xml.j2") }}'
+
+# Start a network
+- virt_net: command=create name=br_nat
+
+# List available networks
+- virt_net: command=list_nets
+
+# Get XML data of a specified network
+- virt_net: command=get_xml name=br_nat
+
+# Stop a network
+- virt_net: command=destroy name=br_nat
+
+# Undefine a network
+- virt_net: command=undefine name=br_nat
+
+# Gather facts about networks
+# Facts will be available as 'ansible_libvirt_networks'
+- virt_net: command=facts
+
+# Gather information about network managed by 'libvirt' remotely using uri
+- virt_net: command=info uri='{{ item }}'
+  with_items: libvirt_uris
+  register: networks
+
+# Ensure that a network is active (needs to be defined and built first)
+- virt_net: state=active name=br_nat
+
+# Ensure that a network is inactive
+- virt_net: state=inactive name=br_nat
+
+# Ensure that a given network will be started at boot
+- virt_net: autostart=yes name=br_nat
+
+# Disable autostart for a given network
+- virt_net: autostart=no name=br_nat
+'''
+
+VIRT_FAILED = 1
+VIRT_SUCCESS = 0
+VIRT_UNAVAILABLE=2
+
+import sys
+
+try:
+    import libvirt
+except ImportError:
+    HAS_VIRT = False
+else:
+    HAS_VIRT = True
+
+try:
+    from lxml import etree
+except ImportError:
+    HAS_XML = False
+else:
+    HAS_XML = True
+
+ALL_COMMANDS = []
+ENTRY_COMMANDS = ['create', 'status', 'start', 'stop',
+                  'undefine', 'destroy', 'get_xml', 'define']
+HOST_COMMANDS = [ 'list_nets', 'facts', 'info' ]
+ALL_COMMANDS.extend(ENTRY_COMMANDS)
+ALL_COMMANDS.extend(HOST_COMMANDS)
+
+ENTRY_STATE_ACTIVE_MAP = {
+    0 : "inactive",
+    1 : "active"
+}
+
+ENTRY_STATE_AUTOSTART_MAP = {
+    0 : "no",
+    1 : "yes"
+}
+
+ENTRY_STATE_PERSISTENT_MAP = {
+    0 : "no",
+    1 : "yes"
+}
+
+class EntryNotFound(Exception):
+    pass
+
+
+class LibvirtConnection(object):
+
+    def __init__(self, uri, module):
+
+        self.module = module
+
+        cmd = "uname -r"
+        rc, stdout, stderr = self.module.run_command(cmd)
+
+        if "xen" in stdout:
+            conn = libvirt.open(None)
+        else:
+            conn = libvirt.open(uri)
+
+        if not conn:
+            raise Exception("hypervisor connection failure")
+
+        self.conn = conn
+
+    def find_entry(self, entryid):
+        # entryid = -1 returns a list of everything
+
+        results = []
+
+        # Get active entries
+        entries = self.conn.listNetworks()
+        for name in entries:
+            entry = self.conn.networkLookupByName(name)
```

Comment:
I suggest that it would be cleaner and more pythonic to do for name in self.conn.listNetworks(): and this kind of code. ( just nitpicking )